packages/grafana-ui/src/components/Layout/LayoutItemContext.ts

```diff
@@ -0,0 +1,21 @@
+import { createContext } from 'react';
+
+export interface LayoutItemContextProps {
+  setAnchoredCount: (nextCount: ((prevCount: number) => number) | number) => void;
```

Comment:
```suggestion
  absoluteElementShown(): () => void;
```

I think something like this would be the easiest to consume, a function that components can call when they want to show some hover element, the function should return function that unregisters it (decrement) .

this way components can basically use it like this 

```ts
useEffect(() => {
   return layoutItemContext.absoluteElementShown();
}, [layoutItemContext])
```

There is no need anything else in the context. Then where the context is wrapped we just need a counter state that is incremented / decremented like this:

```ts
 const [count, setCount] = useState(0);

 const absoluteElementShown = useCallback(() => {
    setCount(count+1);
   return () => setCount(count-1);
 }, [count]));

 const context = useMemo(() => { absoluteElementShown }, [absoluteElementShown]);
```

This could be optimized with a useRef + useState so that the callback is never invalidated and the context value is the same after each call to absoluteElementShown

Reply:
that should work 👍 

the name `absoluteElementShown` feels a bit too generic, imo. we show/hide plenty of absolute elements inside the panels. in fact, we show overflowing tooltips on hover/mousemove where we dont need to rely on this cause it's expensive, and handled well by the `:hover` css rule.

the key difference here is thay we are anchoring a possibly-overflowing child. something like `anchorOverflowingElement()` or `ensureOverflowVisible() is more clear in purpose. or going in the other direction, we can just be explicit/clear about the actual layout effect like `boostZIndex()` or `tempRaiseZIndex()`